source/plugin/optional/basic.rst

```diff
@@ -0,0 +1,195 @@
+
+Thus we can see that allowing implicit nullable contracts leaves us with a selection of pretty awful solutions to
+choose from:
+
+* Plugin authors can assume that **all** methods may return null and code defensively accordingly, however we've already seen that this leads to spaghetti code pretty quickly.
+* The API authors can define an implicit nullable contract on every API method, in an attempt to make null handling the plugin author's problem, which only exacerbates the previous problem.
+* The API authors can assert that any implicit nullable contract they define will never be altered going forward. This means that in the eventuality that they need to handle the removal of a feature from the base game then they must either:
+
+ * Throw an exception - hardly elegant but certainly easier to diagnose than a loose NPE which may be triggered elsewhere in the codebase and be hard to track down
+ * Return a "fake" object or invalid value - this means that consuming (plugin) code will continue to work, but creates an ever-increasing burden on the API developers going forward since every deprecated feature will require the creation of yet more fake objects. This could soon lead to the situation where a big chunk of the API is filled with junk objects whose only purpose is to support parts of the API which are no longer in service.
+
+It should be pretty clear by now that there are some sizable headaches attached to *implicit* nullable contracts, made
+all the more poingant when the API in question is a layer over an extremely unstable base product. Fortunately,
+there is a better way:
+
+2. Optional and the Explicit Nullable Contract
+==============================================
+
+As mentioned above, APIs for Minecraft are in a difficult situation. Ultimately they need to provide a platform with
+a *reasonable amount of implied stability* atop a platform (the game) with *absolutely no amount of implied stability*.
+Thus any API for Minecraft needs to be designed with full awareness that any aspect of the game is liable to change at
+any time for any reason in any way imaginable; up to and including being removed altogether!
+
+This volatility is what leads to the problem with nullable method contracts described above.
+
+`Optional` solves the above problems by replacing *implicit contracts* with *explicit* ones. The API never advertises,
+*"here is your object, kthxbai"*, instead it presents accessors with a
+*"here is a box which may or may not contain the object you asked for, ymmv"*.
+
+.. image:: /images/optionals3.png
+
+By encoding the possibility of returning ``null`` into an explicit contract, we replace the concept of
+*null checking* with the more nuanced concept of *may not exist*. We also stipulate this contract *from day one*.
+
+So what does this mean?
+
+In a nutshell, no longer do plugin authors have to worry about may/may not return ``null``, instead the very
```

Comment:
The wording is still a bit clumsy. Can you either put may/may not return null into a context (or quotes), or perhaps just use "worry about the possibility returning null". Also, do we need the hyperbole of "null-checking the hell out of everything"? Catchy language can make it more readable, but is this really desirable? Just a quibble, ymmv.